workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/Constraint.cs

```diff
@@ -0,0 +1,390 @@
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint Component(uint componentId)
+        {
+            var constraint = Default();
+            constraint.ComponentConstraint = componentId;
+            return new Constraint(constraint);
+        }
+
+        /// <summary>
+        ///     Creates a Constraint object with an And constraint.
+        /// </summary>
+        /// <param name="constraint">
+        ///     First constraint in the list of conjunctions.
+        /// </param>
+        /// <param name="constraints">
+        ///     Further constraints for the list of conjunctions.
+        /// </param>
+        /// <remarks>
+        ///     At least one constraint must be provided to create an "All" QueryConstraint.
+        /// </remarks>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint All(Constraint constraint, params Constraint[] constraints)
+        {
+            return new Constraint(new ComponentInterest.QueryConstraint
+            {
+                AndConstraint = ConstraintParamsToList(constraint, constraints),
+                OrConstraint = new List<ComponentInterest.QueryConstraint>()
+            });
+        }
+
+        /// <summary>
+        ///     Creates a Constraint object with an Or constraint.
+        /// </summary>
+        /// <param name="constraint">
+        ///     First constraint in the list of disjunctions.
+        /// </param>
+        /// <param name="constraints">
+        ///     Further constraints for the list of disjunctions.
+        /// </param>
+        /// <remarks>
+        ///     At least one constraint must be provided to create an "Any" QueryConstraint.
+        /// </remarks>
+        /// <returns>
+        ///     A Constraint object.
+        /// </returns>
+        public static Constraint Any(Constraint constraint, params Constraint[] constraints)
+        {
+            return new Constraint(new ComponentInterest.QueryConstraint
+            {
+                AndConstraint = new List<ComponentInterest.QueryConstraint>(),
+                OrConstraint = ConstraintParamsToList(constraint, constraints)
+            });
+        }
+
+        /// <summary>
+        ///     Returns a QueryConstraint object from a Constraint.
+        /// </summary>
+        /// <returns>
+        ///     A QueryConstraint object.
+        /// </returns>
+        public ComponentInterest.QueryConstraint AsQueryConstraint()
+        {
+            return constraint;
+        }
+
+        private static List<ComponentInterest.QueryConstraint> ConstraintParamsToList(Constraint constraint,
+            params Constraint[] constraints)
+        {
+            var output = new List<ComponentInterest.QueryConstraint>(constraints.Length + 1)
+            {
+                constraint.constraint
+            };
+
+            foreach (var constraintParam in constraints)
```

Comment:
You should be able to use `AddRange` here

Reply:
Don't think we can because we need to convert from the `Constraint` class to the `QueryConstraint` struct here